yarn-project/foundation/src/abi/function_selector.ts

```diff
@@ -97,8 +97,11 @@ export class FunctionSelector {
    * @returns A Buffer containing the 4-byte function selector.
    */
   static fromNameAndParameters(name: string, parameters: ABIParameter[]) {
-    const signature = name === 'constructor' ? name : `${name}(${parameters.map(p => p.type.kind).join(',')})`;
-    return FunctionSelector.fromSignature(signature);
+    const signature = new FunctionSignatureDecoder(name, parameters).decode();
+    const selector = FunctionSelector.fromSignature(signature);
+    // If you are debugging, can be useful to uncomment the following line.
+    // console.log(`Function selector for ${signature} is ${selector}`);
```

Comment:
Why not use a debug logger, so it can be turned on with an env var? Even better, we can have our users turn that on without having to change the code.

Reply:
It probably breaks if you add it as an instance variable, but you can always create a new one as a local variable for this function and then ditch it.